<launch>
  <arg name="DIST"/>
  <arg name="HEIGHT"/>
  <arg name="POSTERIOR"/>
  <arg name="INPUT_IMAGE"/>
  <arg name="MASK"/>
  <arg name="HAND"/>

  <node pkg="jsk_perception" type="colorize_float_image"
      name="$(arg HAND)_colorized_sib_posterior">
  <remap from="~input" to="$(arg POSTERIOR)"/>
  </node>

  <node pkg="jsk_perception" type="colorize_float_image"
      name="$(arg HAND)_colorized_sib_height">
  <remap from="~input" to="$(arg HEIGHT)"/>
  </node>

  <node pkg="jsk_perception" type="colorize_float_image"
      name="$(arg HAND)_colorized_sib_dist">
  <remap from="~input" to="$(arg DIST)"/>
  </node>

  <node pkg="jsk_perception" type="apply_mask_image"
      name="$(arg HAND)_sib_apply_target_mask">
      <remap from="~input" to="$(arg INPUT_IMAGE)"/>
      <remap from="~input/mask" to="$(arg MASK)"/>
      <param name="approximate_sync" value="true"/>
  </node>

  <node name="$(arg HAND)_sib_class_label_image_decomposer"
        pkg="jsk_perception" type="label_image_decomposer.py">
    <remap from="~input" to="/$(arg HAND)_sib_rbo/masked_input" />
    <remap from="~input/label" to="/$(arg HAND)_sib_rbo/class_label" />
  </node>

  <node name="$(arg HAND)_target_mask_to_label"
        pkg="jsk_perception" type="mask_image_to_label.py">
    <remap from="~input" to="/$(arg HAND)_sib_rbo/target_mask" />
  </node>
  <node name="$(arg HAND)_sib_target_label_image_decomposer"
        pkg="jsk_perception" type="label_image_decomposer.py">
    <remap from="~input" to="/$(arg HAND)_softkinetic_camera_resized/image_raw" />
    <remap from="~input/label" to="/$(arg HAND)_target_mask_to_label/output" />
  </node>

  <node name="$(arg HAND)_sib_combine_dist_and_image"
    pkg="jsk_apc2016_common" type="combine_images.py">
    <remap from="~color" to="/$(arg HAND)_softkinetic_camera_resized/image_raw"/>
    <remap from="~mono" to="/$(arg HAND)_sib_spatial/dist"/>
  </node>
</launch>
